<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction fo scMerge2 • scMergeBioc2023</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction fo scMerge2">
<meta property="og:description" content="scMergeBioc2023">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scMergeBioc2023</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/an_intro_to_scMerge2.html">An introduction fo scMerge2</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/yingxinlin/scMergeBioc2023" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction fo scMerge2</h1>
                        <h4 data-toc-skip class="author">Yingxin Lin<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/yingxinlin/scMergeBioc2023/blob/HEAD/vignettes/an_intro_to_scMerge2.Rmd" class="external-link"><code>vignettes/an_intro_to_scMerge2.Rmd</code></a></small>
      <div class="hidden name"><code>an_intro_to_scMerge2.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="atlas-scale-single-cell-multi-sample-multi-condition-data-integration-using-scmerge2">Atlas-scale single-cell multi-sample multi-condition data
integration using scMerge2<a class="anchor" aria-label="anchor" href="#atlas-scale-single-cell-multi-sample-multi-condition-data-integration-using-scmerge2"></a>
</h2>
<p>Authors: Yingxin Lin<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. <br> Last modified: 2 August, 2023.</p>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>The recent emergence of multi-sample multi-condition single-cell
multi-cohort studies allows researchers to investigate different cell
states. The effective integration of multiple large-cohort studies
promises biological insights into cells under different conditions that
individual studies cannot provide.</p>
<div class="section level4">
<h4 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h4>
<p>With the rapid emergence of multi-sample multi-condition single-cell
studies and the increased number of datasets for integration, our
proposed scMerge2 addresses challenges associated with scalability of
cells and studies as well as producing analytically ready data
(i.e. adjusted expression matrix). This is achieved via three key
innovations compared to the previous version of scMerge:</p>
<ol style="list-style-type: decimal">
<li><p>Hierarchical integration is used to capture both local and global
variation. scMerge2 provides users with a more flexible and adaptable
multi-level merging structure, of which each level can comprise multiple
collections of several batches and batch correction can be performed
within each collection separately using user-defined batch
labels.</p></li>
<li><p>Pseudo-bulk construction is used to reduce computing load,
allowing for the analysis of datasets containing millions of
cells.</p></li>
<li><p>Pseudo-replication inside each condition is built, allowing for
the modelling of numerous conditions.</p></li>
</ol>
<p>In essence, scMerge2 takes gene expression matrices from a collection
of datasets and integrates them in a hierarchical manner. The final
output of scMerge2 is a single adjusted expression matrix with all input
data matrices merged and ready for downstream analysis.</p>
</div>
<div class="section level4">
<h4 id="pre-requisites">Pre-requisites<a class="anchor" aria-label="anchor" href="#pre-requisites"></a>
</h4>
<p>It is expected that students will have:</p>
<ol style="list-style-type: decimal">
<li>Basic knowledge of R syntax</li>
<li>Familiarity with <code>SingleCellExperiment</code> objects</li>
</ol>
</div>
<div class="section level4">
<h4 id="participation">Participation<a class="anchor" aria-label="anchor" href="#participation"></a>
</h4>
<p>While it will be possible for participants to run code as we go
through the demonstration, given time constraints, I would encourage
them to focus their attention into integration strategies behind
scMerge2 (pseudo-replicates, pseudo-bulk, stably expressed genes, number
of unwatned variation factors and hierarchical mering etc.). Questions
are welcome both within the workshop and if participants choose to work
through the workshop independently after the demonstration.</p>
</div>
<div class="section level4">
<h4 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h4>
<p>This workshop will focus on Bioconductor packages scMerge and
[SingleCellExperiment] (<a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link uri">https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html</a>).</p>
</div>
<div class="section level4">
<h4 id="time-outline">Time outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h4>
<p>An example for a 45-minute workshop:</p>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction of method</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Introduction of core function</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Hierarchical merging</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Best practise</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="section level2">
<h2 id="workshop">Workshop<a class="anchor" aria-label="anchor" href="#workshop"></a>
</h2>
<div class="section level3">
<h3 id="loading-packages-and-data">Loading Packages and Data<a class="anchor" aria-label="anchor" href="#loading-packages-and-data"></a>
</h3>
<p>We will load the <code>scMerge</code> package. We designed our
package to be consistent with the popular BioConductor’s single cell
analysis framework, namely the <code>SingleCellExperiment</code> and
<code>scater</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/SydneyBioX/scMerge" class="external-link">scMerge</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>We provided an illustrative mouse embryonic stem cell (mESC) data in
our package, as well as a set of pre-computed stably expressed gene
(SEG) list to be used as negative control genes.</p>
<p>The full curated, unnormalised mESC data can be found <a href="http://www.maths.usyd.edu.au/u/yingxinl/wwwnb/scMergeData/sce_mESC.rda" class="external-link">here</a>.
The <code>scMerge</code> package comes with a sub-sampled, two-batches
version of this data (named “batch2” and “batch3” to be consistent with
the full data) .</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Subsetted mouse ESC data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_sce"</span>, package <span class="op">=</span> <span class="st">"scMerge"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"segList_ensemblGeneID"</span>, package <span class="op">=</span> <span class="st">"scMerge"</span><span class="op">)</span></span></code></pre></div>
<p>In this mESC data, we pooled data from 2 different batches from three
different cell types. Using a PCA plot, we can see that despite strong
separation of cell types, there is also a strong separation due to batch
effects. This information is stored in the <code>colData</code> of
<code>example_sce</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_sce</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, exprs_values <span class="op">=</span> <span class="st">"logcounts"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, </span>
<span>                colour_by <span class="op">=</span> <span class="st">"cellTypes"</span>, </span>
<span>                shape_by <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span></code></pre></div>
<p><img src="an_intro_to_scMerge2_files/figure-html/checking%20raw%20data-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="scmerge2">scMerge2<a class="anchor" aria-label="anchor" href="#scmerge2"></a>
</h3>
<div class="section level4">
<h4 id="unsupervised-scmerge2">Unsupervised <code>scMerge2</code><a class="anchor" aria-label="anchor" href="#unsupervised-scmerge2"></a>
</h4>
<p>In unsupervised <code>scMerge2</code>, we will perform graph
clustering on shared nearest neighbour graphs within each batch to
obtain pseudo-replicates. This requires the users to supply a
<code>k_celltype</code> vector with the number of neighbour when
constructed the nearest neighbour graph in each of the batches. By
default, this number is 10.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scMerge2_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scMerge/man/scMerge2.html" class="external-link">scMerge2</a></span><span class="op">(</span>exprsMat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span>,</span>
<span>                         batch <span class="op">=</span> <span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span>,</span>
<span>                         ctl <span class="op">=</span> <span class="va">segList_ensemblGeneID</span><span class="op">$</span><span class="va">mouse</span><span class="op">$</span><span class="va">mouse_scSEG</span>,</span>
<span>                         verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =</span></span>
<span><span class="co">#&gt; TRUE, : You're computing too large a percentage of total singular values, use a</span></span>
<span><span class="co">#&gt; standard svd instead.</span></span>
<span><span class="co">#&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =</span></span>
<span><span class="co">#&gt; TRUE, : You're computing too large a percentage of total singular values, use a</span></span>
<span><span class="co">#&gt; standard svd instead.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">example_sce</span>, <span class="st">"scMerge2"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scMerge2_res</span><span class="op">$</span><span class="va">newY</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2022</span><span class="op">)</span></span>
<span><span class="va">example_sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, exprs_values <span class="op">=</span> <span class="st">'scMerge2'</span><span class="op">)</span>                                       </span>
<span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, colour_by <span class="op">=</span> <span class="st">'cellTypes'</span>, shape <span class="op">=</span> <span class="st">'batch'</span><span class="op">)</span></span></code></pre></div>
<p><img src="an_intro_to_scMerge2_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="semi-supervised-scmerge2">Semi-supervised <code>scMerge2</code><a class="anchor" aria-label="anchor" href="#semi-supervised-scmerge2"></a>
</h4>
<p>When cell type information are known (e.g. results from cell type
classification using reference), scMerge2 can use this information to
construct pseudo-replicates and identify mutual nearest groups with
<code>cellTypes</code> input.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scMerge2_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scMerge/man/scMerge2.html" class="external-link">scMerge2</a></span><span class="op">(</span>exprsMat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span>,</span>
<span>                         batch <span class="op">=</span> <span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span>,</span>
<span>                         cellTypes <span class="op">=</span> <span class="va">example_sce</span><span class="op">$</span><span class="va">cellTypes</span>,</span>
<span>                         ctl <span class="op">=</span> <span class="va">segList_ensemblGeneID</span><span class="op">$</span><span class="va">mouse</span><span class="op">$</span><span class="va">mouse_scSEG</span>,</span>
<span>                         verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">example_sce</span>, <span class="st">"scMerge2"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scMerge2_res</span><span class="op">$</span><span class="va">newY</span></span>
<span></span>
<span><span class="va">example_sce</span> <span class="op">=</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, exprs_values <span class="op">=</span> <span class="st">'scMerge2'</span><span class="op">)</span>                                       </span>
<span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, colour_by <span class="op">=</span> <span class="st">'cellTypes'</span>, shape <span class="op">=</span> <span class="st">'batch'</span><span class="op">)</span></span></code></pre></div>
<p><img src="an_intro_to_scMerge2_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="more-details-of-scmerge2">More details of scMerge2<a class="anchor" aria-label="anchor" href="#more-details-of-scmerge2"></a>
</h3>
<div class="section level4">
<h4 id="number-of-pseudobulk">Number of pseudobulk<a class="anchor" aria-label="anchor" href="#number-of-pseudobulk"></a>
</h4>
<p>The number of pseudobulk constructed within each cell grouping is set
via <code>k_pseudoBulk</code>. By default, this number is set as 30. A
larger number will create more pseudo-bulk data in model estimation,
with longer time in estimation.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scMerge2_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scMerge/man/scMerge2.html" class="external-link">scMerge2</a></span><span class="op">(</span>exprsMat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span>,</span>
<span>                         batch <span class="op">=</span> <span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span>,</span>
<span>                         ctl <span class="op">=</span> <span class="va">segList_ensemblGeneID</span><span class="op">$</span><span class="va">mouse</span><span class="op">$</span><span class="va">mouse_scSEG</span>,</span>
<span>                         k_pseudoBulk <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                         verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =</span></span>
<span><span class="co">#&gt; TRUE, : You're computing too large a percentage of total singular values, use a</span></span>
<span><span class="co">#&gt; standard svd instead.</span></span>
<span><span class="co">#&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =</span></span>
<span><span class="co">#&gt; TRUE, : You're computing too large a percentage of total singular values, use a</span></span>
<span><span class="co">#&gt; standard svd instead.</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">example_sce</span>, <span class="st">"scMerge2"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scMerge2_res</span><span class="op">$</span><span class="va">newY</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2022</span><span class="op">)</span></span>
<span><span class="va">example_sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, exprs_values <span class="op">=</span> <span class="st">'scMerge2'</span><span class="op">)</span>                                       </span>
<span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, colour_by <span class="op">=</span> <span class="st">'cellTypes'</span>, shape <span class="op">=</span> <span class="st">'batch'</span><span class="op">)</span></span></code></pre></div>
<p><img src="an_intro_to_scMerge2_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="return-matrix-by-batch">Return matrix by batch<a class="anchor" aria-label="anchor" href="#return-matrix-by-batch"></a>
</h4>
<p>When working with large data, we can get the adjusted matrix for a
smaller subset of cells each time. This can be achieved by setting
<code>return_matrix</code> to <code>FALSE</code> in
<code><a href="https://rdrr.io/pkg/scMerge/man/scMerge2.html" class="external-link">scMerge2()</a></code> function, which the function then will not
return the adjusted whole matrix but will output the estimated
<code>fullalpha</code>. Then to get the adjusted matrix using the
estimated <code>fullalpha</code>, we first need to performed cosine
normalisation on the logcounts matrix and then calculate the row-wise
(gene-wise) mean of the cosine normalised matrix (This is because by
default, <code><a href="https://rdrr.io/pkg/scMerge/man/scMerge2.html" class="external-link">scMerge2()</a></code> perform cosine normalisation on the
log-normalised matrix before <code>RUVIII</code> step). Then we can use
<code><a href="https://rdrr.io/pkg/scMerge/man/getAdjustedMat.html" class="external-link">getAdjustedMat()</a></code> to adjust the matrix of a subset of cells
each time.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scMerge2_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scMerge/man/scMerge2.html" class="external-link">scMerge2</a></span><span class="op">(</span>exprsMat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span>,</span>
<span>                         batch <span class="op">=</span> <span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span>,</span>
<span>                         ctl <span class="op">=</span> <span class="va">segList_ensemblGeneID</span><span class="op">$</span><span class="va">mouse</span><span class="op">$</span><span class="va">mouse_scSEG</span>,</span>
<span>                         verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                         return_matrix <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =</span></span>
<span><span class="co">#&gt; TRUE, : You're computing too large a percentage of total singular values, use a</span></span>
<span><span class="co">#&gt; standard svd instead.</span></span>
<span><span class="co">#&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =</span></span>
<span><span class="co">#&gt; TRUE, : You're computing too large a percentage of total singular values, use a</span></span>
<span><span class="co">#&gt; standard svd instead.</span></span>
<span></span>
<span><span class="va">cosineNorm_mat</span> <span class="op">&lt;-</span> <span class="fu">batchelor</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/batchelor/man/cosineNorm.html" class="external-link">cosineNorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">adjusted_means</span> <span class="op">&lt;-</span> <span class="fu">DelayedMatrixStats</span><span class="fu">::</span><span class="fu">rowMeans2</span><span class="op">(</span><span class="va">cosineNorm_mat</span><span class="op">)</span></span>
<span></span>
<span><span class="va">newY</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">newY</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scMerge/man/getAdjustedMat.html" class="external-link">getAdjustedMat</a></span><span class="op">(</span><span class="va">cosineNorm_mat</span><span class="op">[</span>, <span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span> <span class="op">==</span> <span class="va">i</span><span class="op">]</span>, </span>
<span>                                <span class="va">scMerge2_res</span><span class="op">$</span><span class="va">fullalpha</span>,</span>
<span>                                ctl <span class="op">=</span> <span class="va">segList_ensemblGeneID</span><span class="op">$</span><span class="va">mouse</span><span class="op">$</span><span class="va">mouse_scSEG</span>,</span>
<span>                                ruvK <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                                adjusted_means <span class="op">=</span> <span class="va">adjusted_means</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">newY</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">cbind</span>, <span class="va">newY</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">example_sce</span>, <span class="st">"scMerge2"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">newY</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2022</span><span class="op">)</span></span>
<span><span class="va">example_sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, exprs_values <span class="op">=</span> <span class="st">'scMerge2'</span><span class="op">)</span>                                       </span>
<span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, colour_by <span class="op">=</span> <span class="st">'cellTypes'</span>, shape <span class="op">=</span> <span class="st">'batch'</span><span class="op">)</span></span></code></pre></div>
<p><img src="an_intro_to_scMerge2_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>Note that we can also adjust only a subset of genes by input a gene
list in <code>return_subset_genes</code> in both
<code><a href="https://rdrr.io/pkg/scMerge/man/getAdjustedMat.html" class="external-link">getAdjustedMat()</a></code> and <code><a href="https://rdrr.io/pkg/scMerge/man/scMerge2.html" class="external-link">scMerge2()</a></code>.</p>
</div>
<div class="section level4">
<h4 id="hierarchical-scmerge2">Hierarchical scMerge2<a class="anchor" aria-label="anchor" href="#hierarchical-scmerge2"></a>
</h4>
<p>scMerge2 provides a hierarchical merging strategy for data
integration that requires multiple level adjustment through function
<code><a href="https://rdrr.io/pkg/scMerge/man/scMerge2h.html" class="external-link">scMerge2h()</a></code>. For example, for the dataset with multiple
samples, we may want to remove the sample effect first within the
dataset before integrate it with other datasets. Below, we will
illustrate how we can build a hierarchical merging order as input for
<code><a href="https://rdrr.io/pkg/scMerge/man/scMerge2h.html" class="external-link">scMerge2h()</a></code>.</p>
<p>For illustration purpose, here I first create a fake sample
information for the sample data. Now, each batch has two samples.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a fake sample information</span></span>
<span><span class="va">example_sce</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">$</span><span class="va">sample</span>, <span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span><span class="op">)</span></span>
<span><span class="co">#&gt;    </span></span>
<span><span class="co">#&gt;     batch2 batch3</span></span>
<span><span class="co">#&gt;   1     50      0</span></span>
<span><span class="co">#&gt;   2     50      0</span></span>
<span><span class="co">#&gt;   3      0     50</span></span>
<span><span class="co">#&gt;   4      0     50</span></span></code></pre></div>
<p>To perform <code>scMerge2h</code>, we need to create</p>
<ol style="list-style-type: decimal">
<li>a hierarchical index list that indicates the indices of the cells
that are going into merging;</li>
<li>a batch information list that indicates the batch information of
each merging.</li>
</ol>
<div class="section level5">
<h5 id="scenario-1">Scenario 1<a class="anchor" aria-label="anchor" href="#scenario-1"></a>
</h5>
<p>We will first illustrate that a two-level merging case, where the
first level refers to the sample effect removal within each batch, and
the second level refers to the merging of two batches.</p>
<p>First, we will construct the hierarchical index list
(<code>h_idx_list</code>). The hierarchical index list is a list that
indicates for each level, which indices of the cells are going into
merging. The number of the element of the list should be the same with
the number of level of merging. For each element, it should contain a
list of vectors of indices of each merging.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Construct a hierarchical index list</span></span>
<span><span class="va">h_idx_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>level1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span><span class="op">)</span>, <span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span><span class="op">)</span>,</span>
<span>                   level2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>On level 1, we will perform two merging, one for each batch.
Therefore, we have a list of two vectors of indices. Each indicates the
indices of the cells of the batches.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h_idx_list</span><span class="op">$</span><span class="va">level1</span></span>
<span><span class="co">#&gt; $batch2</span></span>
<span><span class="co">#&gt;   [1]   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18</span></span>
<span><span class="co">#&gt;  [19]  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36</span></span>
<span><span class="co">#&gt;  [37]  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54</span></span>
<span><span class="co">#&gt;  [55]  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72</span></span>
<span><span class="co">#&gt;  [73]  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90</span></span>
<span><span class="co">#&gt;  [91]  91  92  93  94  95  96  97  98  99 100</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $batch3</span></span>
<span><span class="co">#&gt;   [1] 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118</span></span>
<span><span class="co">#&gt;  [19] 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136</span></span>
<span><span class="co">#&gt;  [37] 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154</span></span>
<span><span class="co">#&gt;  [55] 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172</span></span>
<span><span class="co">#&gt;  [73] 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190</span></span>
<span><span class="co">#&gt;  [91] 191 192 193 194 195 196 197 198 199 200</span></span></code></pre></div>
<p>On level 2, we will perform one merging to merge two batches.
Therefore, we have a list of one vector of indices, indicates all the
indices of the cells of the full data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h_idx_list</span><span class="op">$</span><span class="va">level2</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt;   [1]   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18</span></span>
<span><span class="co">#&gt;  [19]  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36</span></span>
<span><span class="co">#&gt;  [37]  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54</span></span>
<span><span class="co">#&gt;  [55]  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72</span></span>
<span><span class="co">#&gt;  [73]  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90</span></span>
<span><span class="co">#&gt;  [91]  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108</span></span>
<span><span class="co">#&gt; [109] 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126</span></span>
<span><span class="co">#&gt; [127] 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144</span></span>
<span><span class="co">#&gt; [145] 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162</span></span>
<span><span class="co">#&gt; [163] 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180</span></span>
<span><span class="co">#&gt; [181] 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198</span></span>
<span><span class="co">#&gt; [199] 199 200</span></span></code></pre></div>
<p>Next, we need to create the batch information list
(<code>batch_list</code>), which has the same structure with the
<code>h_idx_list</code>, but indicates the batch label of the
<code>h_idx_list</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Construct a batch information list</span></span>
<span><span class="va">batch_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>level1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">$</span><span class="va">sample</span>, <span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span><span class="op">)</span>,</span>
<span>                   level2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">$</span><span class="va">batch</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We can see <code>batch_list</code> indicates the batch label for
level of the merging.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">batch_list</span><span class="op">$</span><span class="va">level1</span></span>
<span><span class="co">#&gt; $batch2</span></span>
<span><span class="co">#&gt;   [1] 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1</span></span>
<span><span class="co">#&gt;  [38] 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2</span></span>
<span><span class="co">#&gt;  [75] 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $batch3</span></span>
<span><span class="co">#&gt;   [1] 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3</span></span>
<span><span class="co">#&gt;  [38] 3 3 3 3 3 3 3 3 3 3 3 3 3 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4</span></span>
<span><span class="co">#&gt;  [75] 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4 4</span></span></code></pre></div>
<p>Now, we can input the <code>batch_list</code> and
<code>h_idx_list</code> in <code>scMerge2h</code> to merge the data
hierarchically. We also need to input a <code>ruvK_list</code>, a vector
of number of unwanted variation (k in RUV model) for each level of the
merging. We suggest a lower <code>ruvK</code> in the first level. Here
we set <code>ruvK_list = c(2, 5)</code> which indicates we will set
RUV’s k equal to 2 in the first level and 5 in the second level.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scMerge2_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scMerge/man/scMerge2h.html" class="external-link">scMerge2h</a></span><span class="op">(</span>exprsMat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span>,</span>
<span>                          batch_list <span class="op">=</span> <span class="va">batch_list</span>,</span>
<span>                          h_idx_list <span class="op">=</span> <span class="va">h_idx_list</span>,</span>
<span>                          ctl <span class="op">=</span> <span class="va">segList_ensemblGeneID</span><span class="op">$</span><span class="va">mouse</span><span class="op">$</span><span class="va">mouse_scSEG</span>,</span>
<span>                          ruvK_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>                          verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The output of <code>scMerge2h</code> is a list of matrices indicates
the adjusted matrix from each level.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">scMerge2_res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">scMerge2_res</span>, <span class="va">dim</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; [1] 1047  200</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; [1] 1047  200</span></span></code></pre></div>
<p>Here we will use the adjusted matrix from the last level as the final
adjusted matrix.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">example_sce</span>, <span class="st">"scMerge2"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scMerge2_res</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">h_idx_list</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2022</span><span class="op">)</span></span>
<span><span class="va">example_sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, exprs_values <span class="op">=</span> <span class="st">'scMerge2'</span><span class="op">)</span>                           </span>
<span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">example_sce</span>, colour_by <span class="op">=</span> <span class="st">'cellTypes'</span>, shape <span class="op">=</span> <span class="st">'batch'</span><span class="op">)</span></span></code></pre></div>
<p><img src="an_intro_to_scMerge2_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="scenario-2">Scenario 2<a class="anchor" aria-label="anchor" href="#scenario-2"></a>
</h5>
<p>scMerge2h can handle a flexible merging strategy input. For example,
on level 1 above, we can only merge data for one batch.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] scater_1.24.0               ggplot2_3.4.0              </span></span>
<span><span class="co">#&gt;  [3] scuttle_1.6.3               scMerge_1.9.99             </span></span>
<span><span class="co">#&gt;  [5] SingleCellExperiment_1.18.1 SummarizedExperiment_1.26.1</span></span>
<span><span class="co">#&gt;  [7] Biobase_2.56.0              GenomicRanges_1.48.0       </span></span>
<span><span class="co">#&gt;  [9] GenomeInfoDb_1.32.4         IRanges_2.30.1             </span></span>
<span><span class="co">#&gt; [11] S4Vectors_0.34.0            BiocGenerics_0.42.0        </span></span>
<span><span class="co">#&gt; [13] MatrixGenerics_1.8.1        matrixStats_0.62.0         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] backports_1.4.1           Hmisc_4.7-1              </span></span>
<span><span class="co">#&gt;   [3] systemfonts_1.0.4         igraph_1.3.5             </span></span>
<span><span class="co">#&gt;   [5] proxyC_0.3.3              splines_4.2.1            </span></span>
<span><span class="co">#&gt;   [7] BiocParallel_1.30.4       inline_0.3.19            </span></span>
<span><span class="co">#&gt;   [9] digest_0.6.30             htmltools_0.5.3          </span></span>
<span><span class="co">#&gt;  [11] viridis_0.6.2             fansi_1.0.3              </span></span>
<span><span class="co">#&gt;  [13] magrittr_2.0.3            checkmate_2.1.0          </span></span>
<span><span class="co">#&gt;  [15] memoise_2.0.1             ScaledMatrix_1.4.1       </span></span>
<span><span class="co">#&gt;  [17] cluster_2.1.3             sfsmisc_1.1-13           </span></span>
<span><span class="co">#&gt;  [19] limma_3.52.4              RcppParallel_5.1.5       </span></span>
<span><span class="co">#&gt;  [21] bdsmatrix_1.3-6           pkgdown_2.0.6            </span></span>
<span><span class="co">#&gt;  [23] prettyunits_1.1.1         jpeg_0.1-9               </span></span>
<span><span class="co">#&gt;  [25] colorspace_2.0-3          ggrepel_0.9.1            </span></span>
<span><span class="co">#&gt;  [27] textshaping_0.3.6         xfun_0.39.1              </span></span>
<span><span class="co">#&gt;  [29] dplyr_1.0.10              callr_3.7.2              </span></span>
<span><span class="co">#&gt;  [31] crayon_1.5.2              RCurl_1.98-1.9           </span></span>
<span><span class="co">#&gt;  [33] jsonlite_1.8.4            survival_3.3-1           </span></span>
<span><span class="co">#&gt;  [35] glue_1.6.2                distr_2.8.0              </span></span>
<span><span class="co">#&gt;  [37] ruv_0.9.7.1               gtable_0.3.1             </span></span>
<span><span class="co">#&gt;  [39] zlibbioc_1.42.0           XVector_0.36.0           </span></span>
<span><span class="co">#&gt;  [41] DelayedArray_0.22.0       pkgbuild_1.4.2           </span></span>
<span><span class="co">#&gt;  [43] BiocSingular_1.12.0       rstan_2.21.7             </span></span>
<span><span class="co">#&gt;  [45] DEoptimR_1.0-11           reldist_1.7-1            </span></span>
<span><span class="co">#&gt;  [47] scales_1.2.1              mvtnorm_1.1-3            </span></span>
<span><span class="co">#&gt;  [49] edgeR_3.38.4              DBI_1.1.3                </span></span>
<span><span class="co">#&gt;  [51] Rcpp_1.0.10               viridisLite_0.4.1        </span></span>
<span><span class="co">#&gt;  [53] htmlTable_2.4.1           dqrng_0.3.0              </span></span>
<span><span class="co">#&gt;  [55] foreign_0.8-82            rsvd_1.0.5               </span></span>
<span><span class="co">#&gt;  [57] Formula_1.2-4             StanHeaders_2.21.0-7     </span></span>
<span><span class="co">#&gt;  [59] ResidualMatrix_1.6.1      metapod_1.4.0            </span></span>
<span><span class="co">#&gt;  [61] htmlwidgets_1.5.4         gplots_3.1.3             </span></span>
<span><span class="co">#&gt;  [63] RColorBrewer_1.1-3        farver_2.1.1             </span></span>
<span><span class="co">#&gt;  [65] pkgconfig_2.0.3           loo_2.5.1                </span></span>
<span><span class="co">#&gt;  [67] nnet_7.3-17               sass_0.4.2               </span></span>
<span><span class="co">#&gt;  [69] deldir_1.0-6              locfit_1.5-9.6           </span></span>
<span><span class="co">#&gt;  [71] utf8_1.2.2                labeling_0.4.2           </span></span>
<span><span class="co">#&gt;  [73] tidyselect_1.2.0          rlang_1.0.6              </span></span>
<span><span class="co">#&gt;  [75] munsell_0.5.0             tools_4.2.1              </span></span>
<span><span class="co">#&gt;  [77] cachem_1.0.6              cli_3.4.1                </span></span>
<span><span class="co">#&gt;  [79] generics_0.1.3            batchelor_1.12.3         </span></span>
<span><span class="co">#&gt;  [81] evaluate_0.17             stringr_1.5.0            </span></span>
<span><span class="co">#&gt;  [83] fastmap_1.1.0             cvTools_0.3.2            </span></span>
<span><span class="co">#&gt;  [85] yaml_2.3.6                ragg_1.2.3               </span></span>
<span><span class="co">#&gt;  [87] processx_3.7.0            knitr_1.42               </span></span>
<span><span class="co">#&gt;  [89] fs_1.5.2                  robustbase_0.95-0        </span></span>
<span><span class="co">#&gt;  [91] caTools_1.18.2            purrr_0.3.5              </span></span>
<span><span class="co">#&gt;  [93] nlme_3.1-157              sparseMatrixStats_1.8.0  </span></span>
<span><span class="co">#&gt;  [95] scran_1.24.1              compiler_4.2.1           </span></span>
<span><span class="co">#&gt;  [97] rstudioapi_0.14           beeswarm_0.4.0           </span></span>
<span><span class="co">#&gt;  [99] png_0.1-7                 tibble_3.1.8             </span></span>
<span><span class="co">#&gt; [101] statmod_1.4.37            bslib_0.4.0              </span></span>
<span><span class="co">#&gt; [103] stringi_1.7.8             highr_0.9                </span></span>
<span><span class="co">#&gt; [105] ps_1.7.1                  desc_1.4.2               </span></span>
<span><span class="co">#&gt; [107] startupmsg_0.9.6          bluster_1.6.0            </span></span>
<span><span class="co">#&gt; [109] lattice_0.20-45           Matrix_1.5-1             </span></span>
<span><span class="co">#&gt; [111] M3Drop_1.22.0             vctrs_0.5.1              </span></span>
<span><span class="co">#&gt; [113] pillar_1.8.1              lifecycle_1.0.3          </span></span>
<span><span class="co">#&gt; [115] jquerylib_0.1.4           BiocNeighbors_1.14.0     </span></span>
<span><span class="co">#&gt; [117] cowplot_1.1.1             data.table_1.14.4        </span></span>
<span><span class="co">#&gt; [119] bitops_1.0-7              irlba_2.3.5.1            </span></span>
<span><span class="co">#&gt; [121] R6_2.5.1                  latticeExtra_0.6-30      </span></span>
<span><span class="co">#&gt; [123] KernSmooth_2.23-20        gridExtra_2.3            </span></span>
<span><span class="co">#&gt; [125] vipor_0.4.5               codetools_0.2-18         </span></span>
<span><span class="co">#&gt; [127] MASS_7.3-57               gtools_3.9.3             </span></span>
<span><span class="co">#&gt; [129] assertthat_0.2.1          rprojroot_2.0.3          </span></span>
<span><span class="co">#&gt; [131] withr_2.5.0               GenomeInfoDbData_1.2.8   </span></span>
<span><span class="co">#&gt; [133] mgcv_1.8-40               parallel_4.2.1           </span></span>
<span><span class="co">#&gt; [135] grid_4.2.1                rpart_4.1.16             </span></span>
<span><span class="co">#&gt; [137] beachmat_2.12.0           densEstBayes_1.0-2.1     </span></span>
<span><span class="co">#&gt; [139] rmarkdown_2.20            DelayedMatrixStats_1.18.2</span></span>
<span><span class="co">#&gt; [141] bbmle_1.0.25              numDeriv_2016.8-1.1      </span></span>
<span><span class="co">#&gt; [143] base64enc_0.1-3           ggbeeswarm_0.6.0         </span></span>
<span><span class="co">#&gt; [145] interp_1.1-3</span></span></code></pre></div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:yingxinlin.res@gmail.com" class="email">yingxinlin.res@gmail.com</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Yale University<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Yingxin Lin.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
